components-core/src/main/java/com/airwallex/android/core/Airwallex.kt

```diff
@@ -183,10 +184,68 @@ class Airwallex internal constructor(
      * Confirm a payment intent with payment consent ID
      *
      * @param session a [AirwallexPaymentSession] used to start the payment flow
-     * @param paymentConsentId the ID of the [PaymentConsent]
+     * @param paymentConsent a [PaymentConsent] used to start the payment flow
      * @param listener The callback of the payment flow
      */
     @UiThread
+    fun confirmPaymentIntent(
+        session: AirwallexPaymentSession,
+        paymentConsent: PaymentConsent,
+        listener: PaymentResultListener
+    ) {
+        val paymentMethod = paymentConsent.paymentMethod
+        val paymentConsentId = paymentConsent.id
+        if (paymentMethod == null) {
+            AirwallexLogger.info("confirmPaymentIntent, paymentMethod == null")
+            listener.onCompleted(AirwallexPaymentStatus.Failure(AirwallexCheckoutException(message = "paymentMethod is required")))
+            return
+        }
+        if (paymentConsentId.isNullOrEmpty()) {
+            AirwallexLogger.info("confirmPaymentIntent, paymentConsentId isNullOrEmpty")
+            listener.onCompleted(AirwallexPaymentStatus.Failure(AirwallexCheckoutException(message = "paymentConsentId is required")))
+            return
+        }
+        if (paymentMethod.card?.numberType == PaymentMethod.Card.NumberType.PAN) {
+            AirwallexLogger.info("confirmPaymentIntent, need cvc")
+            val provider = AirwallexPlugins.getProvider(ActionComponentProviderType.CARD)
+            provider?.get()?.let { paymentProvider ->
+                paymentProvider.handlePaymentData(
+                    AirwallexCheckoutParam(
+                        activity,
+                        paymentMethod,
+                        session,
+                        paymentConsentId
+                    )
+                ) { status: AirwallexPaymentStatus? ->
+                    listener.onCompleted(
+                        status ?: AirwallexPaymentStatus.Failure(
+                            AirwallexCheckoutException(message = "cvc unknown error")
+                        )
+                    )
+                }
+            } ?: run {
+                AirwallexLogger.error("confirmPaymentIntent, Provider is null, unable to handle payment data")
+                listener.onCompleted(
+                    AirwallexPaymentStatus.Failure(
+                        AirwallexCheckoutException(
+                            message = "Cannot find a ComponentProvider with the type ActionComponentProviderType.CARD"
```

Comment:
Might be better align with `Missing ${Dependency.CARD.value} dependency!` in this class so that merchant will know what exactly to do? Also code can be reused.